app/src/hooks/useFileUploader.jsx

```diff
@@ -3,33 +3,47 @@ import { authFetchWithoutContentType } from "../util/url";
 import toast from "react-hot-toast";
 
 const uploadFiles = async (url, { arg }) => {
-  const formData = new FormData();
 
-  Array.from(arg).forEach((file) => {
-    formData.append("files", file);
-  });
+  if (!arg || arg.length === 0) { //if the length of the file array is zero
+    return []; 
+  }
+
+  const results = []; //create empty array to store each backend resposnse
 
-  const response = await authFetchWithoutContentType(url, {
-    method: "POST",
-    body: formData,
-  });
+  for (const file of arg) { //loop over each file
+
+    const formData = new FormData();
+    formData.append("files", file);
 
-  if (!response.ok) {
-    let errorMessage = "File upload failed";
     try {
-      const errorData = await response.json();
-      errorMessage = errorData?.message || response.statusText;
-    } catch (parseError) {
-      errorMessage = response.statusText || "Unknown error";
-      console.log(parseError);
+      const response = await authFetchWithoutContentType(url, { //create a network response for each file
+        method: "POST",
+        body: formData,
+      });
+
+      const rawText = await response.text();
+      let data;
+      try {
+        data = JSON.parse(rawText);
+      } catch (jsonError) {
```

Comment:
We are pretty confident that this will never be reached, but we don't want to completely ignore the possibility of rawText not being proper. I would recommend throwing an error toast and console.error() so we get a Sentry warning.


```suggestion
      } catch (jsonError) {
        toast.error(`Unable to upload ${file.name}`)
        console.error("Unable to upload file", file, jsonError, rawText)
```

Reply:
JSON.parse throws an error. I removed this and just kept it as const data = await response.text(), there is no point of putting in a try and catch.